app/src/main/java/com/mikhailovskii/weatherandroid/data/firebase/FirebaseModel.kt

```diff
@@ -0,0 +1,188 @@
+package com.mikhailovskii.weatherandroid.data.firebase
+
+import com.google.firebase.firestore.FirebaseFirestore
+import com.mikhailovskii.weatherandroid.data.entities.StickerPack
+import com.mikhailovskii.weatherandroid.data.entities.User
+import com.twitter.sdk.android.core.Result
+import timber.log.Timber
+
+class FirebaseModel {
+
+    private val db = FirebaseFirestore.getInstance()
+
+    fun getStickersCollectionList(callback: FirebaseDataCallback?) {
+        db.collection("stickers").get().addOnSuccessListener { result ->
+            val stickerPackList = ArrayList<StickerPack>()
+
+            for (document in result) {
+                val stickerPack = document.toObject(StickerPack::class.java)
+                stickerPackList.add(stickerPack)
+            }
+
+            callback?.onFirebaseDataLoaded(stickerPackList)
+        }.addOnFailureListener {
+            callback?.onFirebaseDataFailed()
+        }
+
+    }
+
+    fun saveUserLocation(user: User?) {
+        db.collection(USERS_COLLECTION).whereEqualTo(LOGIN_FIELD, user?.login).get()
+            .addOnSuccessListener { databaseResult ->
+
+                val map: Map<String, String> = hashMapOf(LOCATION to (user?.location ?: ""))
+
+                for (document in databaseResult) {
+                    db.collection(USERS_COLLECTION).document(document.id).update(map)
+                }
+
+            }
+    }
+
+    fun baseLogInUser(login: String, password: String, callback: FirebaseLoginCallback?) {
```

Comment:
@SergeiMikhailovskii probably it would be better to be consistent with the other `logIn` methods and call it `logInWithPassword`